src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -177,6 +182,13 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                         // can also occur if the consumer calls it directly and just doesn't pass a constant value.
                         regNumber baseReg = node->ExtractTempReg();
                         regNumber offsReg = node->GetSingleTempReg();
+
+                        // Ensure we aren't overwriting op1Reg or op2Reg
+                        assert(baseReg != op1Reg);
+                        assert(baseReg != op2Reg);
+                        assert(offsReg != op1Reg);
+                        assert(offsReg != op2Reg);
```

Comment:
IMO these assertions are a bit of overkill, but I guess they can't hurt.

Reply:
I had initially added them because I was unsure if `tmpReg` can ever be `opReg` (https://github.com/dotnet/coreclr/pull/16649#discussion_r171142486).

I am going to remove these "unneeded" asserts as part of the rebase.